service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
> So IIUC, making the function return instanceID and then, whenever we need to get information about the instance, call DescribeInstances() with instanceID is what we should do. Right?

Yes.

> This is what the PR did at the beginning ...

Ok cool. Then lets go with what we had initially. I was missing some information back then.